### ISSUE TITLE
Performance problem on devel (`update` and `draw` called twice per loop, probably)

### ISSUE THREAD

**Issue by AndrewRadev** [ggez/ggez#861]
**Describe the bug**

On the current `devel` branch, I'm getting a slowdown. I think the problem is with #774, particularly this snippet: https://github.com/ggez/ggez/pull/774/files#diff-14da7c350de5742eaa248de110fefeae9b4b639c0358e26024d6040ccf174a41R309-R310. I think the `update` and `draw` functions are being called a second time in the loop.

On the latest `devel` commit there's a todo on top, so maybe it's a known problem, but I couldn't see a recent open issue about it (though I've only looked at the titles, sorry if this is a duplicate): https://github.com/ggez/ggez/blob/427661476d3d2133d74c7e7d73b6c6bc0a1cd40e/src/event.rs#L310-L311

Commenting those two lines out seems to fix my problem.

**To Reproduce**

I'm using the style of main loop shown in the examples, nothing fancier:

``` rust
const DESIRED_FPS: u32 = 60;

while timer::check_update_time(ctx, DESIRED_FPS)? {
    let seconds = 1.0 / (DESIRED_FPS as f32);
```

The project I'm testing with is a simple game for teaching purposes, tutorial-ish: https://github.com/AndrewRadev/rust-shooter/tree/update-ggez-2020. But I can reproduce it with astroblasto, too. When the cursor hovers on top of the window, the whole game freezes up.

**Expected behavior**

Tested on 5bbb5c37ea04eaa80f75480a2e977c0752c45d14 (before #774):

https://user-images.githubusercontent.com/124255/103132824-4d000480-46af-11eb-8d32-6b64060345d1.mp4

**Actual behaviour**

Tested on 1d9abaccae574e69b49693f03e269b980c210bd9 (after #774):

https://user-images.githubusercontent.com/124255/103132833-65701f00-46af-11eb-8943-2eacb8c24162.mp4

**Hardware and Software:**
 - ggez version: current git devel branch
 - OS: Linux (Arch), X11
 - Graphics card: Embedded Intel, also happens on Radeon RX 640, doesn't seem like it should matter if I'm right about the issue

**Comment by nobbele** [comment]
that snippet is the problem, removing the last update/draw call fixes it without any issues. please fix upstream

**Comment by icefoxen** [comment]
Fixed in #862, thank you.

**Comment by AndrewRadev** [comment]
Awesome, thanks :+1:
